stl/src/locale0.cpp

```diff
@@ -157,9 +188,21 @@ _MRTIMP2_PURE locale::_Locimp* __CLRCALL_PURE_OR_CDECL locale::_Init(bool _Do_in
         ptr->_Catmask = all; // set current locale to "C"
         ptr->_Name    = "C";
 
-        _Locimp::_Clocptr = ptr; // set classic to match
-        _Locimp::_Clocptr->_Incref();
-        ::new (&classic_locale) locale(_Locimp::_Clocptr);
+        // set classic to match
+        ptr->_Incref();
+        ::new (&classic_locale) locale(ptr);
+#if defined(_M_CEE_PURE)
+        locale::_Locimp::_Clocptr = ptr;
+#else // ^^^ _M_CEE_PURE / !_M_CEE_PURE vvv
+        const auto mem = reinterpret_cast<volatile intptr_t*>(&locale::_Locimp::_Clocptr);
+        const auto as_bytes = reinterpret_cast<intptr_t>(ptr);
+        _Compiler_or_memory_barrier();
+#ifdef _WIN64
+        __iso_volatile_store64(mem, as_bytes);
+#else
+        __iso_volatile_store32(mem, as_bytes);
+#endif

```

Comment:
Comment nitpick: As these are nested, I would prefer to see the `#else` and `#endif` commented.